app/src/main/java/org/splitbrain/giraffe/EventLoader.java

```diff
@@ -101,25 +180,51 @@ protected String doInBackground(URL... urls) {
             //AssetManager assetManager = context.getAssets();
             //InputStream inputStream = assetManager.open("rp2011.ics");
 
-            SimpleIcalParser ical = new SimpleIcalParser(inputStream);
-            SimpleIcalEvent event;
-            while ((event = ical.nextEvent()) != null) {
-                // build record
-                EventRecord record = getEventRecord(event);
-                if (record == null) continue;
-
-                // add to database:
-                db.addEventRecord(record);
-                count++;
-
-                // progress feedback
-                publishProgress("Loaded " + count + " events...");
-
-                // abort if cancelled
-                if (isCancelled()) {
-                    db.rollback();
-                    db.close();
-                    return "Cancelled";
+            if (type.equals("ical")) {
+                SimpleIcalParser ical = new SimpleIcalParser(inputStream);
+                SimpleIcalEvent event;
+                while ((event = ical.nextEvent()) != null) {
+                    // build record
+                    EventRecord record = getEventRecordIcal(event);
+                    if (record == null) continue;
+
+                    // add to database:
+                    db.addEventRecord(record);
+                    count++;
+
+                    // progress feedback
+                    publishProgress("Loaded " + count + " events...");
+
+                    // abort if cancelled
+                    if (isCancelled()) {
+                        db.rollback();
+                        db.close();
+                        return "Cancelled";
+                    }
+                }
+            } else if (type.equals("json")) {
+                SimpleJsonParser json = new SimpleJsonParser(output);
+                ArrayList<SimpleJsonEvent> event = json.nextEvent();
+                int i;
+
+                for (i = 0; i < event.size(); i++) {
+                    // build record
+                    EventRecord record = getEventRecordJson(event.get(i));
+                    if (record == null) continue;
+
+                    // add to database:
+                    db.addEventRecord(record);
+                    count++;
+
+                    // progress feedback
+                    publishProgress("Loaded " + count + " events...");
+
+                    // abort if cancelled
+                    if (isCancelled()) {
+                        db.rollback();
+                        db.close();
+                        return "Cancelled";
+                    }
```

Comment:
There is too much duplicated code in this if construct. Both branches do effectively the same. They only differ in the used getRecord function and how the list of events is read. This should be reflected as such here.